src/structures/NewsChannel.js

```diff
@@ -13,6 +14,25 @@ class NewsChannel extends TextChannel {
     // News channels don't have a rate limit per user, remove it
     this.rateLimitPerUser = undefined;
   }
+
+  /**
+   * Adds the target to this channel's followers.
+   * @param {GuildChannelResolvable} channel The channel where the webhook should be created
+   * @param {string} [reason] Reason for creating the webhook
+   * @returns {Promise<NewsChannel>}
+   * @example
+   * if (channel.type === 'news') {
+   *   channel.addFollower('222197033908436994', 'Important announcements')
+   *     .then(() => console.log('Added follower'))
+   *     .catch(console.error);
+   * }
+   */
+  async addFollower(channel, reason) {
+    channel = this.client.channels.resolve(channel);
+    if (!channel || !['text', 'news'].includes(channel.type)) throw new Error('GUILD_TEXT_CHANNEL_RESOLVE');
```

Comment:
I don't think that it will work if the channel type is `text`. Why aren't you just checking if the type is `news` ?

Reply:
Regular text channels can also follow announcement channels, that's not restricted to only the `news` type

Reply:
I see, but why don't you use the `isText` fonction to know if a channel is text based ? It would support any future text based channel type

Reply:
because that method on the other hand also supports dm channels, which can't follow announcement channels